Add Progress thumb size

GDmail91 opened the pull request:
I want to set thumb size of each circular progress bars.
So I splitted the thumb size and progress thickness.

I've added default thumb size (as 10dp) and thumb & shadow Paint object.

---
I cannot find other branch. So I pushed in master branch.

GuilhE commented:
Hello @GDmail91 and thanks for your contribution.
It's a nice feature but you forgot to handle 2 corner cases:

1. `onMeasure`:

![Screenshot 2020-07-26 at 14 07 10](https://user-images.githubusercontent.com/2677139/88479774-7b838300-cf49-11ea-90b3-01a61cfa8b72.png)

2. Compatibility

A flag `autoThumb` (`true` by default): the thumb changes it's size according to progress thickness.
If `autoThumb` is `false`, user is responsible for adjusting the thumbSize independently progressThickness.
This way we won't break compatibility.

GuilhE commented:
Hello again @GDmail91 , thanks for you changes.
We're almost done, just 2 minor changes:
1. Remove **Thumb.java** since it's not being used;
2. Please add` { } `in **if-else** conditions, example:
 ```
if (mProgressThumbScaleType == POINT)
    mProgressThumbSize = size;
else
    mProgressThumbSize = Math.max(Math.min(size, mMaxThumbSizeRate), 0); // To prevent the Thumb size too big
```
to:
```
 if (mProgressThumbScaleType == POINT) {
     mProgressThumbSize = size;
 } else {
     mProgressThumbSize = Math.max(Math.min(size, mMaxThumbSizeRate), 0); // To prevent the Thumb size too big
 }
```

Thanks.

GDmail91 commented:
I Fixed  above issues and also auto & rate scaleType calculate issue

You can see the whole feature that I've added in sample app

Here is some description

1. Changed progress thumb size 

ProgressThumbScaleType.AUTO : It will be same as before
ProgressThumbScaleType.POINT : It will change the Thumb size by dp 
ProgressThumbScaleType.RATE : It will change the Thumb size by rate of stroke

Default Progress Scale Type is AUTO

```xml
... 
<com.github.guilhe.views.CircularProgressView
                android:id="@+id/sample_CircularProgressView"
                android:layout_width="@dimen/progress_size"
                android:layout_height="@dimen/progress_size"
                app:layout_constraintEnd_toEndOf="parent"
                app:layout_constraintStart_toStartOf="parent"
                app:layout_constraintTop_toTopOf="parent"
                app:progress="100"
                app:progressThumbScaleType="point"
                app:progressThumbSize="35dp"
                app:progressBarColorArray="@array/rainbow"
                app:progressBarRounded="true"
                app:progressThumb="false" />


```

GDmail91 commented:
+ 
I think there is no rate size setting by xml 
I will added it soon

GuilhE commented:
> I think there is no rate size setting by xml
> I will added it soon

Ok thanks!
I'll add some comments in code where you missed `{ }` 😉

GuilhE commented on sample/src/main/java/sample/SampleActivity.java:
```diff
@@ -95,6 +104,14 @@ public void onProgressChanged(SeekBar seekBar, int progress, boolean fromUser) {
                 break;
+            case R.id.thumbsize_SeekBar:
+                float size;
+                if (mBinding.sampleCircularProgressView.getProgressThumbScaleType() == ProgressThumbScaleType.RATE)
```
Please wrap in `{ }`

GuilhE commented on circular-progress-view/src/main/java/com/github/guilhe/views/CircularProgressView.java:
```diff
@@ -604,7 +676,19 @@ protected synchronized void onDraw(Canvas canvas) {
+            radius -= (thumbSize - mProgressStrokeThickness / 2);
+        } else if (mProgressThumbScaleType == RATE) {
+            thumbSize = (mProgressStrokeThickness / 2) * mProgressThumbSize;
+            if (mProgressThumbSize > 1)
```
@GDmail91 Please wrap in `{ }`

GuilhE commented on circular-progress-view/src/main/java/com/github/guilhe/views/CircularProgressView.java:
```diff
@@ -572,11 +623,32 @@ protected synchronized void onMeasure(int widthMeasureSpec, int heightMeasureSpe
+            if (mProgressThumbScaleType == POINT && thumbSize * 2 > mProgressStrokeThickness) {
+                progressWidth += thumbSize - mProgressStrokeThickness + mProgressStrokeThickness / 2;
+            } else if (mProgressThumbScaleType == RATE) {
+                if (thumbSize > mProgressStrokeThickness)
```
@GDmail91 Please wrap in `{ }`

GuilhE commented:
Thanks for your contribution @GDmail91  👍
